### ISSUE TITLE
YOLO11-pose detection, low point confidence at the image edge

### ISSUE THREAD

**Issue by qq97017090** [ultralytics/ultralytics#22431]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

In keypoint detection tasks, I only need to detect one class with 4 keypoints per instance. The shape formed by these 4 keypoints is approximately a rectangle, with two points always located at the image edges. During training, I found that keypoints in the central region of the image have confidence scores above 0.7, while those at the edges have confidence scores below 0.3. Due to hardware limitations, I can only use the 's' model with an image size of 96*160. I suspect this phenomenon occurs because fewer features are available at the image edges. Therefore, when creating the dataset, I've tried to keep the edge points at a certain distance from the actual image boundaries. Are there other possible causes for this issue and any additional solutions?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @qq97017090, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started—an Ultralytics engineer will follow up here shortly to assist further.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including:
-Exact Ultralytics version (`ultralytics` package), OS, Python, and PyTorch versions
-YOLO11 model variant used and training command/args
-Complete training logs and results (including any warnings)
-Dataset details (number of images, keypoint format, sample images with labels)
-Image size, augmentations, and hyperparameters used
-Validation metrics and qualitative examples (images where confidence differs center vs. edge)
This will help us better understand and reproduce your observations 🙂

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
It's probably low because it's at the edge of the bounding box. There should be some margin between the bounding box and the keypoints.

**Comment by ErwinCheung** [comment]
The image edge distortion is severe, and the target becomes smaller. The difficulty of detection has increased.  
A large amount of real data that appears on distorted edge targets needs to be supplemented.
At the same time, do more affine data augmentation

**Comment by glenn-jocher** [comment]
Good observations above—two additional factors to check: during augmentation any keypoint pushed outside the image is marked not visible and excluded from loss, which can depress confidence near borders, and at 96×160 the feature map resolution is very coarse at the edges. To mitigate: reduce/disable geometric aug that causes out‑of‑frame keypoints, turn off cut‑paste augments, switch to `yolo11n-pose.pt` and raise `imgsz` as high as memory allows, and ensure your pose YAML defines `kpt_shape` and a correct `flip_idx` so flips don’t swap keypoint identities.

```bash
yolo train task=pose model=yolo11n-pose.pt data=your_pose.yaml imgsz=192 \
  mosaic=0.0 mixup=0.0 perspective=0.0 translate=0.05 degrees=0 shear=0 scale=0.10 \
  fliplr=0.0 flipud=0.0
```

For reference, keypoints outside the image are zeroed in the augmenter’s `apply_keypoints` step, and the affine/perspective operations that can cause this are detailed in the augmenter docs; also note the plotter may hide points exactly on the border when visualizing. See the `apply_keypoints` behavior and available transforms in the augmentation reference, the affine transform description, and plotting notes for keypoints in the results/plotting references: [apply_keypoints in data augmentation](https://docs.ultralytics.com/reference/data/augment/#apply_keypoints), [affine/perspective pipeline](https://docs.ultralytics.com/reference/data/augment/#affine_transform), [Keypoints/plotting details](https://docs.ultralytics.com/reference/utils/plotting/#kpts) and the [YOLO pose task guide](https://docs.ultralytics.com/tasks/pose/).
